testutil/ssntp_test_certs.go

```diff
@@ -486,3 +495,123 @@ func RoleToTestCert(role ssntp.Role) string {
 
 	return TestCertUnknown
 }
+
+// RoleToTestCertPath returns a string containing the path to a test
+// ca cert file, a string containing the path to the cert file for
+// the given role
+func RoleToTestCertPath(role ssntp.Role) (string, string, error) {
+	certdir, err := os.Getwd()
+	if err != nil {
+		return "", "", errors.New("Unable to get current directory")
+	}
+	cacert := path.Join(certdir, "CAcert-localhost.pem")
+	roleCert := ""
+
+	switch role {
+	case ssntp.SCHEDULER:
+		roleCert = path.Join(certdir, "cert-Scheduler-localhost.pem")
+	case ssntp.AGENT:
+		roleCert = path.Join(certdir, "cert-CNAgent-localhost.pem")
+	case ssntp.Controller:
+		roleCert = path.Join(certdir, "cert-Controller-localhost.pem")
+	case ssntp.CNCIAGENT:
+		roleCert = path.Join(certdir, "cert-CNCIAgent-localhost.pem")
+	case ssntp.NETAGENT:
+		roleCert = path.Join(certdir, "cert-NetworkingAgent-localhost.pem")
+	}
+
+	if roleCert == "" {
+		err = errors.New("No cert for role")
+	}
+
+	return cacert, roleCert, err
+}
+
+func makeTestCert(role ssntp.Role) error {
+	caPath, path, err := RoleToTestCertPath(role)
+	if err != nil {
+		return fmt.Errorf("Missing cert path: %v", err)
+	}
+
+	template, err := certs.CreateCertTemplate(role, "test", "test@test.test", []string{"localhost"}, []string{"127.0.0.1"})
+	if err != nil {
+		return fmt.Errorf("Unable to create cert template: %v", err)
+	}
+
+	writer, err := os.Create(path)
+	if err != nil {
+		return fmt.Errorf("Unable to create cert file: %v", err)
+	}
+	defer writer.Close()
```

Comment:
So I want to note that I changed this but it is really cumbersome and fragile needing to have all the cleanup done at every return point. Hopefully I'm missing a go idiom here but without exceptions or goto or automatic cleanup of some sort I don't see a particularly better way to write it. @tpepper I'd love to see thoughts on this part when you review.